File: ios/LittleChef/LittleChef/Models/Recipe.swift
import Foundation

struct Recipe: Identifiable {
    let id: UUID
    let title: String
    let ingredients: [String: String]
    let steps: [String]
    let sourceUrl: String
    let images: [String?]
    let isSaved: Bool
    let servings: Int
    let totalTime: Int?
    
    var mainImage: String? {
        images.first ?? nil
    }
    
    var ingredientCount: Int {
        ingredients.count
    }
    
    var stepCount: Int {
        steps.count
    }
}

extension RecipeDTO {
    func toRecipe() -> Recipe {
        Recipe(
            id: id,
            title: title,
            ingredients: ingredients,
            steps: steps,
            sourceUrl: sourceUrl,
            images: images,
            isSaved: isSaved,
            servings: servings ?? 4,  // Default to 4 servings if not provided by backend
            totalTime: totalTime
        )
    }
}

File: backend/app/schemas/recipe.py
from pydantic import BaseModel
from typing import List, Optional
from uuid import UUID

class Recipe(BaseModel):
    id: UUID
    title: str
    ingredients: dict
    steps: List[str]
    source_url: Optional[str] = None
    images: List[Optional[str]] = []
    total_time: Optional[int] = None
    servings: Optional[int] = 4  # Default to 4 servings if not provided
    tags: List[str] = []
    hash: str
    is_saved: bool = False

    class Config:
        from_attributes = True